Examples/Io/EDM4hep/src/EDM4hepUtil.cpp

```diff
@@ -0,0 +1,305 @@
+
+  // Only used for direction; normalization/units do not matter
+  to.setDirection(from.getMomentum()[0], from.getMomentum()[1],
+                  from.getMomentum()[2]);
+
+  to.setAbsoluteMomentum(std::hypot(from.getMomentum()[0],
+                                    from.getMomentum()[1],
+                                    from.getMomentum()[2]) *
+                         Acts::UnitConstants::GeV);
+
+  return to;
+}
+
+void EDM4hepUtil::writeParticle(const ActsFatras::Particle& from,
+                                edm4hep::MutableMCParticle to) {
+  // TODO what about particleId?
+
+  to.setPDG(from.pdg());
+  to.setCharge(from.charge() / Acts::UnitConstants::e);
+  to.setMass(from.mass() / Acts::UnitConstants::GeV);
+  to.setVertex({from.position().x(), from.position().y(), from.position().z()});
+  to.setMomentum({(float)from.fourMomentum().x(),
+                  (float)from.fourMomentum().y(),
+                  (float)from.fourMomentum().z()});
+}
+
+ActsFatras::Hit EDM4hepUtil::readSimHit(const edm4hep::SimTrackerHit& from,
+                                        MapParticleIdFrom particleMapper,
+                                        MapGeometryIdFrom geometryMapper) {
+  ActsFatras::Barcode particleId = particleMapper(from.getMCParticle());
+
+  // TODO what about the digitization?
+  Acts::GeometryIdentifier geometryId = geometryMapper(from.getCellID());
+
+  const auto mass = from.getMCParticle().getMass();
+  const Acts::ActsVector<3> momentum{
+      from.getMomentum().x * Acts::UnitConstants::GeV,
+      from.getMomentum().y * Acts::UnitConstants::GeV,
+      from.getMomentum().z * Acts::UnitConstants::GeV,
+  };
+  const auto energy = std::hypot(momentum.norm(), mass);
+
+  ActsFatras::Hit::Vector4 pos4{
+      from.getPosition().x * Acts::UnitConstants::mm,
+      from.getPosition().y * Acts::UnitConstants::mm,
+      from.getPosition().z * Acts::UnitConstants::mm,
+      from.getTime() * Acts::UnitConstants::ns,
+  };
+
+  ActsFatras::Hit::Vector4 mom4{
+      momentum.x(),
+      momentum.y(),
+      momentum.z(),
+      energy,
+  };
+
+  // TODO no EDM4hep equivalent?
+  ActsFatras::Hit::Vector4 delta4{
+      0 * Acts::UnitConstants::GeV, 0 * Acts::UnitConstants::GeV,
+      0 * Acts::UnitConstants::GeV,
+      0 * Acts::UnitConstants::GeV,  // sth.getEDep()
+  };
+
+  // TODO no EDM4hep equivalent?
+  int32_t index = -1;
+
+  return ActsFatras::Hit(geometryId, particleId, pos4, mom4, mom4 + delta4,
+                         index);
+}
+
+void EDM4hepUtil::writeSimHit(const ActsFatras::Hit& from,
+                              edm4hep::MutableSimTrackerHit to,
+                              MapParticleIdTo particleMapper,
+                              MapGeometryIdTo geometryMapper) {
+  const Acts::Vector4& globalPos4 = from.fourPosition();
+  const Acts::Vector4& momentum4Before = from.momentum4Before();
+  const auto delta4 = from.momentum4After() - momentum4Before;
+
+  if (particleMapper != nullptr) {
```

Comment:
I **think** `operator bool()` is well defined for `std::function`. So maybe

```cpp
if (particleMapper) {}
```

is enough?